benchmarking/simulation/simulation.py

```diff
@@ -104,9 +129,10 @@ def run(self, num_runs: int) -> SimulationArtifact:
 
 def run_trial_batch(
     fragment: int,
-    num_runs_for_batch: int,
+    run_indexes_for_batch: List[int],
```

Comment:
I don't think this should get to this level. I think this function should only have awareness of the num runs and the seeds it needs

We should have already only passed the seeds at the needed indexes to this function, essentiallyh